From 8cdef3521cfccd9cb796db63cd19df9dd90bd374 Mon Sep 17 00:00:00 2001
From: Mark Grover <mark@apache.org>
Date: Fri, 8 Apr 2016 10:10:10 -0700
Subject: [PATCH 184/201] [SPARK-14477][BUILD] Allow custom mirrors for
 downloading artifacts in build/mvn

Allows to override locations for downloading Apache and Typesafe artifacts in build/mvn script.

By running script like
````
rm -rf build/apache-maven*
rm -rf build/zinc-*
rm -rf build/scala-*

...

build/mvn -Pyarn -Phadoop-2.4 -Dhadoop.version=2.6.0 package

APACHE_MIRROR=http://mirror.infra.cloudera.com/apache/ build/mvn -Pyarn -Phadoop-2.4 -Dhadoop.version=2.6.0 package

APACHE_MIRROR=http://mirror.infra.cloudera.com/apache build/mvn -Pyarn -Phadoop-2.4 -Dhadoop.version=2.6.0 package

APACHE_MIRROR=xyz build/mvn -Pyarn -Phadoop-2.4 -Dhadoop.version=2.6.0 package
````

Author: Mark Grover <mark@apache.org>

Closes #12250 from markgrover/spark-14477.

(cherry picked from commit a9b630f42ac0c6be3437f206beddaf0ef737f5c8)
(cherry picked from commit 19aac558badce163a12fc4e4ee10246daa76b9ae)
---
 build/mvn |   12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/build/mvn b/build/mvn
index 7603ea0..55146e0 100755
--- a/build/mvn
+++ b/build/mvn
@@ -71,8 +71,11 @@ install_app() {
 install_mvn() {
   local MVN_VERSION="3.3.3"
 
+  local MVN_VERSION="3.3.9"
+  local APACHE_MIRROR=${APACHE_MIRROR:-https://archive.apache.org/dist}
+
   install_app \
-    "http://archive.apache.org/dist/maven/maven-3/${MVN_VERSION}/binaries" \
+    "${APACHE_MIRROR}/maven/maven-3/${MVN_VERSION}/binaries" \
     "apache-maven-${MVN_VERSION}-bin.tar.gz" \
     "apache-maven-${MVN_VERSION}/bin/mvn"
 
@@ -83,8 +86,10 @@ install_mvn() {
 install_zinc() {
   local zinc_path="zinc-0.3.5.3/bin/zinc"
   [ ! -f "${_DIR}/${zinc_path}" ] && ZINC_INSTALL_FLAG=1
+  local TYPESAFE_MIRROR=${TYPESAFE_MIRROR:-https://downloads.typesafe.com}
+
   install_app \
-    "http://downloads.typesafe.com/zinc/0.3.5.3" \
+    "${TYPESAFE_MIRROR}/zinc/0.3.5.3" \
     "zinc-0.3.5.3.tgz" \
     "${zinc_path}"
   ZINC_BIN="${_DIR}/${zinc_path}"
@@ -98,9 +103,10 @@ install_scala() {
   local scala_version=`grep "scala.version" "${_DIR}/../pom.xml" | \
                        head -1 | cut -f2 -d'>' | cut -f1 -d'<'`
   local scala_bin="${_DIR}/scala-${scala_version}/bin/scala"
+  local TYPESAFE_MIRROR=${TYPESAFE_MIRROR:-https://downloads.typesafe.com}
 
   install_app \
-    "http://downloads.typesafe.com/scala/${scala_version}" \
+    "${TYPESAFE_MIRROR}/scala/${scala_version}" \
     "scala-${scala_version}.tgz" \
     "scala-${scala_version}/bin/scala"
 
-- 
1.7.9.5

